core/jazz_environment-event-handler/index.js

```diff
@@ -597,6 +601,265 @@ function triggerBuildJob(payload, serviceDetails, configData) {
+      method: "POST",
+      headers: {
+        "Authorization": authToken,
+        "Content-Type": "application/json",
+        "Jazz-Service-ID": service_id
+      },
+      json: updatePayload,
+      rejectUnauthorized: false
+    };
+    request(svcPayload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200) {
+        const timestamp = new Date().toISOString();
+        const vaultLink = configData.TVAULT.HOSTNAME + "/#!/admin";
+        environmentPayload['safe_details'] = { "name": safeName, "ts": timestamp, "link": vaultLink };
+        safeDetails = environmentPayload['safe_details'];
+        return resolve(environmentPayload);
+      } else {
+        logger.error("Error creating safe: " + JSON.stringify(response));
+        return reject({
+          "error": "Error creating safe",
+          "details": response.body.message
+        });
+      }
+    });
+
+  });
+}
+
+function deleteSafe(environmentPayload, configData, authToken, envDetails) {
+  return new Promise((resolve, reject) => {
+    if (!configData.TVAULT && !configData.TVAULT.IS_ENABLED) {
+      return resolve({
+        "error": "T-vault is not enabled",
+      });
+    }
+    let safeName;
+    if (envDetails.data && envDetails.data.count > 0 && envDetails.data.environment.length){
+      const envResponse = envDetails.data.environment.filter(ele => ele.logical_id === environmentPayload.logical_id);
+      if (envResponse.length && envResponse[0].metadata && envResponse[0].metadata.safe_details) {
+        safeName = envResponse[0].metadata.safe_details.name;
+      } else {
+        return resolve({
+          "error": environmentPayload.logical_id + ": environment safe details is not found",
+        });
+      }
+    } else {
+      return resolve({
+        "error": environmentPayload.logical_id + ": environment safe details is not found",
+      });
+    }
+    var svcPayload = {
+      uri: configData.BASE_API_URL + '/' + environmentPayload.logical_id + configData.TVAULT.API + '/' +safeName,
+      method: "DELETE",
+      headers: {
+        "Authorization": authToken,
+        "Content-Type": "application/json",
+      },
+      rejectUnauthorized: false
+    };
+    request(svcPayload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200) {
+        return resolve(body);
+      } else {
+        logger.error("Error deleting safe: " + JSON.stringify(response));
+        return reject({
+          "error": "Error deleting safe",
+          "details": response.body.message
+        });
+      }
+    });
+
+  });
+}
+
+function getAdmins(environmentPayload, configData, authToken) {
+  return new Promise((resolve, reject) => {
+    var payload = {
+      uri: configData.BASE_API_URL + '/' + environmentPayload.logical_id + configData.TVAULT.USERS,
```

Comment:
same as above comment. Please see in other places also if applicable

Reply:
Removed logical_id and addressed.